internal/registration/internal/domain/user_registration.go

```diff
@@ -19,26 +66,39 @@ type UserRegistration struct {
 	ConfirmationDate time.Time
 }
 
-func RegisterNewUser(
+func NewUserRegistrationSnapshot(
```

Comment:
I cannot see the purpose of this function. Why you need to create a snapshot with new id and time.Now?

Perhaps this is extra.